src/settings/LocationLibraries.js

```diff
@@ -32,97 +27,43 @@ const translations = {
-      params: {
-        query: 'cql.allRecords=1 sortby name',
-        limit: '10000',
-      },
-      accumulate: true,
-    },
-  });
 
-  static propTypes = {
-    intl: PropTypes.object,
-    stripes: PropTypes.shape({
-      connect: PropTypes.func.isRequired,
-      hasPerm: PropTypes.func.isRequired,
-    }).isRequired,
-    resources: PropTypes.shape({
-      institutions: PropTypes.object,
-      campuses: PropTypes.object,
-      locationsPerLibrary: PropTypes.object,
-    }).isRequired,
-    mutator: PropTypes.shape({
-      institutions: PropTypes.shape({
-        GET: PropTypes.func.isRequired,
-        reset: PropTypes.func.isRequired,
-      }),
-      campuses: PropTypes.shape({
-        GET: PropTypes.func.isRequired,
-        reset: PropTypes.func.isRequired,
-      }),
-      locationsPerLibrary: PropTypes.shape({
-        GET: PropTypes.func.isRequired,
-        reset: PropTypes.func.isRequired,
-      }),
-    }),
-  };
+const LocationLibraries = (props) => {
+  const stripes = useStripes();
+  const { formatMessage } = useIntl();
+  const [institutionId, setInstitutionId] = useState(sessionStorage.getItem(INSTITUTION_ID_LIBRARIES) || null);
+  const [campusId, setCampusId] = useState(sessionStorage.getItem(CAMPUS_ID_LIBRARIES) || null);
 
-  constructor(props) {
-    super(props);
-    this.connectedControlledVocab = props.stripes.connect(ControlledVocab);
-    this.hasAllLocationPerms = props.stripes.hasPerm('ui-tenant-settings.settings.location');
-    this.numberOfObjectsFormatter = this.numberOfObjectsFormatter.bind(this);
+  const hasAllLocationPerms = stripes.hasPerm('ui-tenant-settings.settings.location');
+  const ConnectedControlledVocab = stripes.connect(ControlledVocab);
 
-    this.state = {
-      institutionId: null,
-      campusId: null,
-    };
-  }
+  const { institutions } = useInstitutions({ searchParams: {
+    limit: 100,
+    query: 'cql.allRecords=1 sortby name',
+  } });
 
-  /**
-   * Refresh lookup tables when the component mounts. Fetches in the manifest
-   * will only run once (in the constructor) but because this object may be
-   * unmounted/remounted without being destroyed/recreated, the lookup tables
-   * will be stale if they change between unmounting/remounting.
-   */
-  componentDidMount() {
-    const institutionId = sessionStorage.getItem(INSTITUTION_ID_LIBRARIES);
-    const campusId = sessionStorage.getItem(CAMPUS_ID_LIBRARIES);
-    this.setState({ institutionId, campusId });
-    ['institutions', 'campuses', 'locationsPerLibrary'].forEach(i => {
-      this.props.mutator[i].reset();
-      this.props.mutator[i].GET();
-    });
-  }
+  const { campuses } = useCampuses({ searchParams: {
+    limit: 100,
+    query: 'cql.allRecords=1 sortby name',
+  } });
+
+  const { locations } = useLocations({ searchParams: {
+    limit: 10000,
+    query: 'cql.allRecords=1 sortby name',
+  } });
+
+  useEffect(() => {
+    sessionStorage.setItem(INSTITUTION_ID_LIBRARIES, institutionId);
```

Comment:
Suggestion: What do you think about encapsulating working with  NSTITUTION_ID_LIBRARIES and CAMPUS_ID_LIBRARIES (state, changing, and connecting to local storage) in separate hooks and just return API for that?